plugins/woocommerce/src/Admin/API/AI/Product.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\API\AI;
+
+use Automattic\WooCommerce\Blocks\AIContent\UpdateProducts;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product controller
+ *
+ * @internal
+ */
+class Product {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'ai';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/product',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'update_product' ),
+					'permission_callback' => array( Middleware::class, 'is_authorized' ),
+					'args'                => array(
+						'products_information' => array(
+							'description' => __( 'Data generated by AI for updating dummy products.', 'woocommerce' ),
+							'type'        => 'object',
+						),
+						'last_product'         => array(
+							'description' => __( 'Whether the product being updated is the last one in the loop', 'woocommerce' ),
+							'type'        => 'boolean',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Update product with the content and image powered by AI.
+	 *
+	 * @param  WP_REST_Request $request Request object.
+	 *
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_product( WP_REST_Request $request ) {
```

Comment:
You're right, done here dd3c58673c73b46e018e7ffdd1d63721deebb899, and also updated the Patterns one since it was trying to catch a `WPError` instead of an `Exception`.